bson/bsoncodec/struct_tag_parser.go

```diff
@@ -117,3 +121,19 @@ var DefaultStructTagParser StructTagParserFunc = func(sf reflect.StructField) (S
 
 	return st, nil
 }
+
+// JSONFallbackStructTagParser has the same behavior as DefaultStructTagParser
+// but will also fallback to parsing the json tag instead on a field where the
+// bson tag isn't available.
+var JSONFallbackStructTagParser StructTagParserFunc = func(sf reflect.StructField) (StructTags, error) {
+	key := strings.ToLower(sf.Name)
+	tag, ok := sf.Tag.Lookup("bson")
+	if !ok {
+		tag, ok = sf.Tag.Lookup("json")
+	}
+	if !ok && !strings.Contains(string(sf.Tag), ":") && len(sf.Tag) > 0 {
```

Comment:
Is this fully tested? I think we're missing 2 tests:

1. We only enter this case if the `bson` tag is not present (or if both `bson` and `json` tags are not present when using the fallback parser)
2. We only use this tag if it does not have a `:`. For example, we should not honor `xml` tags.

Reply:
tags without key:value aren't actually legal (I think we accommodate them because of mgo) and reflect.StructField won't parse tags that mix un-keyed and keyed pairs, so i didn't add tests for it.